### ISSUE TITLE
Segmentation fault while running noisy_circle.py example on GPU

### ISSUE THREAD

**Issue by manishsaroya** [bruel-gabrielsson/TopologyLayer#15]
I am getting "Segmentation fault (core dumped)" when I try to run noisy_circle.py on GPU.
I just changed this part of code in `levelset/noisy_circle.py`
```
device = torch.device('cuda')
tloss = TopLoss((50,50)).to(device) # topology penalty
dloss = nn.MSELoss().to(device) # data loss

beta_t = torch.autograd.Variable(torch.tensor(beta_ols, device=device).type(torch.float), requires_grad=True)
X_t = torch.tensor(X, dtype=torch.float, requires_grad=False)
y_t = torch.tensor(y, dtype=torch.float, requires_grad=False)
optimizer = torch.optim.Adam([beta_t], lr=1e-2)
```
The error occurs when ```X.extendFloat(f)``` is called in `sublevel.py`. 
Some version details.
```
nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2017 NVIDIA Corporation
Built on Fri_Nov__3_21:07:56_CDT_2017
Cuda compilation tools, release 9.1, V9.1.85

$ gcc --version
gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
``` 
Can you please help me with this.

Thanks

**Comment by bnels** [comment]
@manishsaroya I've put in a pull request for a fix.  Basically, this happened because the persistence calculation happens on CPU, but you passed in a memory address on GPU.  I have some further comments in #16.

If you want a quick fix, try (in the optimization loop)
```python
tlossi = tloss(beta_t.cpu()).to(device)
```
if the pull request goes through then you don't need to do the above

You will also want (perhaps) to put `X_t` and `y_t` on the device
```python
X_t = torch.tensor(X, dtype=torch.float, requires_grad=False, device=device)
y_t = torch.tensor(y, dtype=torch.float, requires_grad=False, device=device)
```

**Comment by bnels** [comment]
Closing since I believe this issue has now been fixed in #16

**Comment by manishsaroya** [comment]
Yeah, it works now. Thanks
